crates/project/src/project.rs

```diff
@@ -1552,6 +1552,21 @@ impl Project {
         })
     }
 
+    pub fn copy_relative_entry(
```

Comment:
Great, thank you for finding the right method combination to copy things with.
Let's avoid extending project's public API for such small things.

I think we can remove this `pub fn` from the `project.rs` and either unite with the `fn create_relative_path` or have a separate function/method that does both.
It feels rather small to live without a separate function though.

Reply:
In personly, I like modify exists api `worktree::copy_entry` to add an new parameters, like
```diff
    pub fn copy_entry(
        &mut self,
        entry_id: ProjectEntryId,
++        relative_path: Option<PathBuf>,
        new_path: impl Into<Arc<Path>>,
        cx: &mut ModelContext<Self>,
    ) -> Task<Result<Option<Entry>>> {
```